src/Controls/src/Core/Shell/ShellContent.cs

```diff
@@ -63,6 +63,17 @@ Page IShellContentController.GetOrCreateContent()
 			}
 			else
 			{
+				if (template.Type != null)
+				{
+					template.LoadTemplate = () =>
+					{
+						if (Parent?.FindMauiContext()?.Services != null)
+						{
+							return Extensions.DependencyInjection.ActivatorUtilities.GetServiceOrCreateInstance(Parent.FindMauiContext().Services, template.Type);
```

Comment:
```suggestion
                        var services = Parent?.FindMauiContext()?.Services;
						if (services != null)
						{
							return Extensions.DependencyInjection.ActivatorUtilities.GetServiceOrCreateInstance(services, template.Type);
```

Reply:
@pictos Oddly, I did try this some time ago, and very weirdly `services` always ended up as `null` even when `Parent`, `FindMautiContext()` and `Services` all returned values. I stared at it in the debugger for a long time and decided not to fight with it so things ended up as you see now. 
I think I can potentially chalk that up to preview weirdness, and will try it again as you suggest.

Reply:
@PureWeen do you have any idea about this?

Reply:
My guess is that you were probably seeing some weirdness with the IDE, the debugger wasn't being truthful, or for some reason your libraries weren't actually recompiled because of the IDE. 

Let's see if it works the second time around :-)